### PR TITLE
Feature/tokenomics ramm swap integration tests

### PR DESCRIPTION
**Pull request by rackstar**
## Context

Add RAMM integration tests.

## Changes proposed in this pull request

* add RAMM contract to integration/setup.js
* add RAMM swap integration tests

## Test plan

Please describe the tests cases that you ran to verify your changes. Add further instructions on 
how to run them if needed (i.e. migration / deployment scripts, env vars, etc).

## Checklist

- [ ] Rebased the base branch
- [ ] Attached corresponding Github issue
- [ ] Prefixed the name with the type of change (i.e. feat, chore, test)
- [ ] Performed a self-review of my own code
- [ ] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [ ] Didn't generate new warnings
- [ ] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -18,7 +18,7 @@
   },
   "ignorePatterns": ["node_modules/", "artifacts", "coverage", "!.solcover.js"],
   "parserOptions": {
-    "ecmaVersion": 2018
+    "ecmaVersion": 2020
```
Comment (rackstar) [inline .eslintrc:21]: bumping to 2020 to support optional chaining

File: test/integration/Ramm/swap.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  const fixture = await loadFixture(setup);
+  const { tk, p1, tc } = fixture.contracts;
+  const [member1] = fixture.accounts.members;
+
+  const operator = await tk.operator();
+  await setEtherBalance(operator, parseEther('10000'));
+  await setEtherBalance(member1.address, parseEther('10000'));
+  await setEtherBalance(p1.address, parseEther('145000'));
+
+  await tk.connect(await ethers.getImpersonatedSigner(operator)).mint(member1.address, parseEther('1000000000000'));
```
Comment (shark0der) [inline test/integration/Ramm/swap.js:29]: Won't this drop the BV significantly?

Comment (rackstar) [reply test/integration/Ramm/swap.js:29]: i didn't thought of that. to be honest, the value came from existing code in integration tests. I have now set it to a more conservative value of `10_000`

File: test/integration/Ramm/swap.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    // +3 hours to reach below BV, this ensures price is stable between the 2 swaps below
+    const nextBlockTimestamp = timestamp + 3 * 60 * 60;
+    const deadline = nextBlockTimestamp + 5 * 60; // add 5 minutes
+
+    // Get expected book value
+    const initState = await ra.loadState();
+    const capital = await p1.getPoolValueInEth();
+    const supply = await tc.totalSupply();
+    const mcrValue = await mcr.getMCR();
+    const state = await ra._getReserves(initState, capital, supply, mcrValue, nextBlockTimestamp);
```
Comment (rackstar) [inline test/integration/Ramm/swap.js:124]: wasted a lot of time trying to get `ra.getReserves()` to work.

getReserves always has a different `block.timestamp` to what I set in `setNextBlockTime`, not really sure why.

so i ended up using the dirtier way of `ra._getReserves`.

**Review by shark0der — APPROVED** [review]

**Pull request merged**